lte/gateway/c/oai/tasks/mme_app/mme_app_context.c

```diff
@@ -1450,38 +1450,47 @@ void mme_ue_context_update_ue_sig_connection_state(
     if (mme_config.nas_config.t3412_min > 0) {
       // Start Mobile reachability timer only if peroidic TAU timer is not
       // disabled
-      nas_itti_timer_arg_t timer_callback_arg = {0};
-      timer_callback_arg.nas_timer_callback =
-          mme_app_handle_mobile_reachability_timer_expiry;
-      timer_callback_arg.nas_timer_callback_arg =
-          (void*) &(ue_context_p->mme_ue_s1ap_id);
-      if (timer_setup(
-              ue_context_p->mobile_reachability_timer.sec, 0, TASK_MME_APP,
-              INSTANCE_DEFAULT, TIMER_ONE_SHOT, &timer_callback_arg,
-              sizeof(timer_callback_arg),
-              &(ue_context_p->mobile_reachability_timer.id)) < 0) {
-        OAILOG_ERROR_UE(
-            LOG_MME_APP, ue_context_p->emm_context._imsi64,
-            "Failed to start Mobile Reachability timer for UE id "
-            " " MME_UE_S1AP_ID_FMT "\n",
-            ue_context_p->mme_ue_s1ap_id);
-        ue_context_p->mobile_reachability_timer.id = MME_APP_TIMER_INACTIVE_ID;
+      if (ue_context_p->mobile_reachability_timer.id ==
```

Comment:
@VinashakAnkitAman , can you explain why is there a dangling mobile reachability timer in the first place? If the UE is switching to ECM_IDLE state, it should not have any existing mobile reachability timer. This condition should always be true.

Reply:
This check is added for additional safety to not restart this timer and log the information if it is already started, in worst case of not cleaning the timers in connection with any previous test case of Sanity where cleanups are not done properly or UE state is not updated properly. Although this place was creating an additional dangling pointer, but the other check that I have added in line 1430 is sufficient in fixing our crash. 

For the test cases where mobile reachability timer was getting started but not stopped, during Reset and shutdown event, this function was called 2nd time to update the UE signalling state and without checking the current state we were starting the timer here looking on the new state only. So I have added a check in line 1430 here to check current state also and that itself is fixing the crash for mobile reachability timer expiry handling.

Reply:
Thanks for the clarification @VinashakAnkitAman !